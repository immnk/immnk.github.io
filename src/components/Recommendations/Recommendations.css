.root {
}

.container {
  width: 720px;
  margin: 0 auto;
  padding: 8rem 2.4rem;

  @media screen and (max-width: 540px) {
    width: 100%;
  }
}

.header {
  font-family: 'Abel', sans-serif;
  font-weight: 800;
  font-size: 3.2rem;
  margin-bottom: 2.4rem;
  text-align: center;
}

.description {
  font-weight: 300;
  line-height: 1.5;
  font-size: 1.6rem;
  text-align: center;
}

.list {
  margin-top: 2rem;
}

.recommendation {
  font-size: 1.4rem;
  display: flex;
  margin-bottom: 2rem;
  border: 1px solid #1111111f;
  padding: 10px;
}

.avatar {
  width: 100px;
  height: 100px;
  border-radius: 50%;
  object-fit: cover;
}

.name {
  font-size: 1.6rem;
  text-align: center;
}

.text {
  flex: 1;
  margin-left: 2rem;
}
