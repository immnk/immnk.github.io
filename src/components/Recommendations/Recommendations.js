import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import ScrollAnimation from 'react-animate-on-scroll';

import YuryImage from './images/yury.jpeg';
import MarkImage from './images/mark.jpeg';
import ChintanImage from './images/chintan.jpeg';
import VamsiImage from './images/vamsi.jpeg';

import css from './Recommendations.css';

const Recommendations = ({ className }) => {
  const recommendations = [
    {
      name: 'Yury',
      image: YuryImage,
      text: `I worked with Manikanta for almost two years 
      and he shown himself to be a capable and productive developer. 
      He's a quick learner, inquisitive and resourceful. 
      He can start a new package and get it live, he can independently develop a new infrastructure tool, 
      and is always on the lookout for new challenges.`
    },
    {
      name: 'Mark',
      image: MarkImage,
      text: `Manikanta is a talented developer. He is an adventurer and not afraid to venture new technologies. 
        He thinks fast and provide valuable insights for the team. He is a guy that gets the job done.`
    },
    {
      name: 'Chintan',
      image: ChintanImage,
      text:
        'Highly skilled developer. Codes very well and of high quality. He was our go to guy when he was in Sirius.'
    },
    {
      name: 'Vamsi',
      image: VamsiImage,
      text: `I have worked with Manikanta, for several different projects ranging 
        from windows mobile apps to android to angular and node.js. 
        He is an excellent problem solver, he has no fear for diving into the code bases written by others, 
        regardless of complexity. Hacker by nature, he quickly gives you the prototype of your app 
        and iterate over it by taking the feedback. He has a good visual sense like a designer, 
        and he also know sketch, posts random things in behance and things like that.
         He has passion to complete the product with in the stipulated time, with the best of the quality code and UI. 
         I look forward to work with him for the multiple projects in the days to come.`
    }
  ];
  return (
    <section className={classNames(css.root, className)}>
      <ScrollAnimation animateIn="fadeIn" delay={100} initiallyVisible={false}>
        <div className={css.container}>
          <h3 className={css.header}>Testimonials I received!</h3>
          <p className={css.description}>
            All recommendations/Testimonials are fetched from my{' '}
            <a href="https://www.linkedin.com/in/manikantatankala/">LinkedIn</a> profile.
          </p>
          <div className={css.list}>
            {recommendations.map((r) => {
              return (
                <div className={css.recommendation} key={r.name}>
                  <div>
                    <img className={css.avatar} src={r.image} alt={r.name} />
                    <h5 className={css.name}>{r.name}</h5>
                  </div>
                  <p className={css.text}>{r.text}</p>
                </div>
              );
            })}
          </div>
        </div>
      </ScrollAnimation>
    </section>
  );
};

Recommendations.defaultProps = {
  className: undefined
};

Recommendations.propTypes = {
  className: PropTypes.string
};

export default Recommendations;
