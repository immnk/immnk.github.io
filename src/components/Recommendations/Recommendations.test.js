import React from 'react';
import { shallow } from 'enzyme';

import Recommendations from './Recommendations';

const renderRecommendations = (props) => {
  const defaultProps = {
    className: 'my-classname'
  };

  return <Recommendations {...defaultProps} {...props} />;
};

const shallowRenderRecommendations = (props) => shallow(renderRecommendations(props));

describe('Test Recommendations', () => {
  it('render app', () => {
    const actual = shallowRenderRecommendations();
    expect(actual).toMatchSnapshot();
  });
});
