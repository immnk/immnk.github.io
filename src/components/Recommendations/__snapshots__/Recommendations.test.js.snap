// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Recommendations render app 1`] = `
<section
  className="root my-classname"
>
  <ScrollAnimation
    animateIn="fadeIn"
    animateOnce={false}
    animatePreScroll={true}
    delay={100}
    duration={1}
    initiallyVisible={false}
    offset={150}
  >
    <div
      className="container"
    >
      <h3
        className="header"
      >
        Testimonials I received!
      </h3>
      <p
        className="description"
      >
        All recommendations/Testimonials are fetched from my
         
        <a
          href="https://www.linkedin.com/in/manikantatankala/"
        >
          LinkedIn
        </a>
         profile.
      </p>
      <div
        className="list"
      >
        <div
          className="recommendation"
          key="Yury"
        >
          <div>
            <img
              alt="Yury"
              className="avatar"
              src="test-file-stub"
            />
            <h5
              className="name"
            >
              Yury
            </h5>
          </div>
          <p
            className="text"
          >
            I worked with Manikanta for almost two years 
      and he shown himself to be a capable and productive developer. 
      He's a quick learner, inquisitive and resourceful. 
      He can start a new package and get it live, he can independently develop a new infrastructure tool, 
      and is always on the lookout for new challenges.
          </p>
        </div>
        <div
          className="recommendation"
          key="Mark"
        >
          <div>
            <img
              alt="Mark"
              className="avatar"
              src="test-file-stub"
            />
            <h5
              className="name"
            >
              Mark
            </h5>
          </div>
          <p
            className="text"
          >
            Manikanta is a talented developer. He is an adventurer and not afraid to venture new technologies. 
        He thinks fast and provide valuable insights for the team. He is a guy that gets the job done.
          </p>
        </div>
        <div
          className="recommendation"
          key="Chintan"
        >
          <div>
            <img
              alt="Chintan"
              className="avatar"
              src="test-file-stub"
            />
            <h5
              className="name"
            >
              Chintan
            </h5>
          </div>
          <p
            className="text"
          >
            Highly skilled developer. Codes very well and of high quality. He was our go to guy when he was in Sirius.
          </p>
        </div>
        <div
          className="recommendation"
          key="Vamsi"
        >
          <div>
            <img
              alt="Vamsi"
              className="avatar"
              src="test-file-stub"
            />
            <h5
              className="name"
            >
              Vamsi
            </h5>
          </div>
          <p
            className="text"
          >
            I have worked with Manikanta, for several different projects ranging 
        from windows mobile apps to android to angular and node.js. 
        He is an excellent problem solver, he has no fear for diving into the code bases written by others, 
        regardless of complexity. Hacker by nature, he quickly gives you the prototype of your app 
        and iterate over it by taking the feedback. He has a good visual sense like a designer, 
        and he also know sketch, posts random things in behance and things like that.
         He has passion to complete the product with in the stipulated time, with the best of the quality code and UI. 
         I look forward to work with him for the multiple projects in the days to come.
          </p>
        </div>
      </div>
    </div>
  </ScrollAnimation>
</section>
`;
